tools/sof_perf_analyzer.py

```diff
@@ -169,8 +165,13 @@ def process_trace_file():
     '''The top-level caller for processing the trace file'''
     with open(args.filename, 'r', encoding='utf8') as file:
         trace_item_gen = make_trace_item(file)
-        trace_prev = skip_to_first_trace(trace_item_gen)
-        if trace_prev is None:
+        trace_prev = None
+        try:
+            if args.skip_to_first_trace:
+                trace_prev = skip_to_first_trace(trace_item_gen)
+            else:
+                trace_prev = next(trace_item_gen)
+        except:
```

Comment:
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html

Why do you need to catch anything here? Just let the original exception propagate to the top. This is test code, we want as few abstractions and layers as possible when things go wrong.

Reply:
The raised exception has no useful information, I want to add some useful information here, I will use below code, which is better, add message to the exception, and raise it again.
```
        try:
            if args.skip_to_first_trace:
                trace_prev = skip_to_first_trace(trace_item_gen)
            else:
                trace_prev = next(trace_item_gen)
        except StopIteration as si:
            si.args = 'No valid trace in provided file',
            raise 
```